src/Infrastructure/PiwikCustomTracking.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend\Infrastructure;
+
+/**
+ * Simple wrapper for tracking methods in Piwik's API
+ *
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class PiwikCustomTracking {
+
+	const SCOPE_VISIT = 'visit';
+	const SCOPE_PAGE = 'page';
+
+	const EVENT_SET_CUSTOM_VARIABLE = 'setCustomVariable';
+	const EVENT_TRACK_GOAL = 'trackGoal';
+
+	private $events = [];
+
+	public function setCustomVariable( int $variableId, string $name, string $value, string $scope ) {
+		$this->events[] = [ self::EVENT_SET_CUSTOM_VARIABLE, $variableId, $name, $value, $scope ];
+	}
+
+	public function trackGoal( int $goalId ) {
+		$this->events[] = [ self::EVENT_TRACK_GOAL, $goalId ];
+	}
+
+	public function getEvents(): array {
+		return $this->events;
+	}
```

Comment:
There is some not entirely obvious binding here. `getEvents` only user is a thing where the result gets shoved into a template. When first looking at this class, that was not clear to me at all. Callers of the two event recording methods are thus using the same format as the template. That's very suspicious.